packages/core/core/src/Graph.js

```diff
@@ -422,7 +422,12 @@ export default class Graph<TNode: Node, TEdgeType: string | null = null> {
         }
       }
 
-      if (typeof visit !== 'function' && visit.exit) {
+      if (
+        typeof visit !== 'function' &&
+        visit.exit &&
+        // Make sure the graph still has the node: it may have been removed between enter and exit
```

Comment:
This patch uncovered an issue where we were attempting to visit (on exit) a node after it had been removed between enter and exit, so check here first.